File: fjt.pricingservice/MySqlDBModel/rfq_consolidated_mfgpn_lineitem.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Collections.Generic;

namespace fjt.pricingservice.MySqlDBModel
{
    [Table("rfq_consolidated_mfgpn_lineitem")]
    public class rfq_consolidated_mfgpn_lineitem
    {
        public rfq_consolidated_mfgpn_lineitem()
        {
            rfq_consolidate_mfgpn_lineitem_quantity = new HashSet<rfq_consolidate_mfgpn_lineitem_quantity>();
            rfq_consolidated_mfgpn_lineitem_alternate = new HashSet<rfq_consolidated_mfgpn_lineitem_alternate>();
            //rfq_lineitem_autopricingstatus = new HashSet<rfq_lineitem_autopricingstatus>();
        }
        [Key]
        public int id { get; set; }
     //   public int rfqAssyBomID { get; set; }
        [StringLength(8)]
        public string lineID { get; set; }
        public bool? isInstall { get; set; }
        public bool? isPurchase { get; set; }
        //public decimal? qty { get; set; }
        public int? rfqAssyID { get; set; }
        public int? partTypeID { get; set; }
        public int? partClassID { get; set; }
        public decimal? leadQty { get; set; }
        public decimal? attritionRate { get; set; }
        public bool? isActive { get; set; }
        public int? rfqLineItemID { get; set; }
        public int? uomID { get; set; }
        public bool? isNoBidsPN { get; set; }
        public bool? isDraft { get; set; }
        public decimal? leads { get; set; }
        public decimal? totalQty { get; set; }
        public decimal? requestQty { get; set; }
        public decimal? originalTotalQty { get; set; }
        public double? qpa { get; set; }
        public DateTime createdAt { get; set; }
        public DateTime updatedAt { get; set; }
        public DateTime? deletedAt { get; set; }
        public string createdBy { get; set; }
        public string updatedBy { get; set; }
        public string deletedBy { get; set; }
        public bool? isDeleted { get; set; }
        public bool isMultiple { get; set; }
        public decimal? valueAddedCost { get; set; }
        public decimal? numOfPosition { get; set; }
        public int? numOfRows { get; set; }
   //     public virtual rfq_assy_bom rfq_assy_bom { get; set; }
        public virtual rfq_assemblies rfq_assemblies { get; set; }
        public virtual ICollection<rfq_consolidate_mfgpn_lineitem_quantity> rfq_consolidate_mfgpn_lineitem_quantity { get; set; }
        public virtual ICollection<rfq_consolidated_mfgpn_lineitem_alternate> rfq_consolidated_mfgpn_lineitem_alternate { get; set; }
        //public virtual ICollection<rfq_lineitem_autopricingstatus> rfq_lineitem_autopricingstatus { get; set; }
    }
}

File: fjt.api/models/RFQConsolidatedMFGPNLineItem.js
const Sequelize = require('sequelize');

module.exports = (sequelize) => {
    const RFQConsolidatedMFGPNLineItem = sequelize.define('RFQConsolidatedMFGPNLineItem', {

        id: {
            type: Sequelize.INTEGER,
            allowNull: false,
            autoIncrement: true,
            primaryKey: true
        },
        lineID: {
            type: Sequelize.STRING,
            allowNull: true
        },
        isInstall: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        isPurchase: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        rfqAssyID: {
            type: Sequelize.INTEGER,
            allowNull: false
        },
        rfqLineItemID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        uomID: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        totalQty: {
            type: Sequelize.DECIMAL(8, 2),
            allowNull: true
        },
        requestQty: {
            type: Sequelize.DECIMAL(8, 2),
            allowNull: true
        },
        originalTotalQty: {
            type: Sequelize.DECIMAL(8, 2),
            allowNull: true
        },
        isMultiple: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        numOfPosition: {
            type: Sequelize.DECIMAL(8, 2),
            allowNull: true
        },
        numOfRows: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        qpa: {
            type: Sequelize.DECIMAL(16, 8),
            allowNull: true
        },
        isdeleted: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        createdBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        updatedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        deletedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        deletedAt: {
            type: Sequelize.DATE,
            allowNull: true
        },
        isqpaMismatch: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        consolidatedpartlineID: {
            type: Sequelize.STRING,
            allowNull: true
        },
        consolidatedPartWithFlagDetail: {
            type: Sequelize.STRING,
            allowNull: true
        }

    }, {
        paranoid: true,
        tableName: 'rfq_consolidated_mfgpn_lineitem'
    });

    RFQConsolidatedMFGPNLineItem.associate = (models) => {
        RFQConsolidatedMFGPNLineItem.belongsTo(models.RFQAssemblies, {
            as: 'rfqAssemblies',
            foreignKey: 'rfqAssyID'
        });
        RFQConsolidatedMFGPNLineItem.belongsTo(models.RFQAssyQuantity, {
            as: 'rfqAssyQuantity',
            foreignKey: 'qtyID'
        });
        RFQConsolidatedMFGPNLineItem.hasMany(models.RFQConsolidatedMFGPNLineItemAlternate, {
            as: 'rfqConsolidatedMFGPNLineItemAlternate',
            foreignKey: 'consolidateID'
        });
        RFQConsolidatedMFGPNLineItem.hasMany(models.RFQConsolidatedMFGPNLineItemQuantity, {
            as: 'rfqConsolidatedMFGPNLineItemQuantity',
            foreignKey: 'consolidateID'
        });

        RFQConsolidatedMFGPNLineItem.belongsTo(models.RFQLineItems, {
            as: 'rfqLineItems',
            foreignKey: 'rfqLineItemID'
        });

        RFQConsolidatedMFGPNLineItem.belongsTo(models.UOMs, {
            as: 'uoms',
            foreignKey: 'uomID'
        });
    };

    return RFQConsolidatedMFGPNLineItem;
};
